### PR TITLE
Support array indexing for duckdb

### PR DESCRIPTION
**Pull request by JichaoS**
https://duckdb.org/docs/sql/data_types/list#retrieving-from-lists

### CODE REVIEW

File: tests/sqlparser_duckdb.rs [hunk 1]
```diff
@@ -516,3 +516,37 @@ fn test_duckdb_named_argument_function_with_assignment_operator() {
...
+    let sql = r#"SELECT ['a', 'b', 'c'][3] AS three"#;
+    let stmt = duckdb().verified_stmt(sql);
+    let query = match stmt {
+        Statement::Query(q) => q,
+        _ => panic!("Expected a query"),
+    };
+    let select = match *query.body {
+        SetExpr::Select(s) => s,
+        _ => panic!("Expected a select"),
+    };
```
Comment (jmhain) [inline tests/sqlparser_duckdb.rs:531]:
```suggestion
    let select = duckdb().verified_only_select(sql);
```

**Review by jmhain — APPROVED** [review]
lgtm, thanks! cc @alamb

File: tests/sqlparser_duckdb.rs (hunk 1 above)
Comment (JichaoS) [reply tests/sqlparser_duckdb.rs]: fixed

**Review by alamb — APPROVED** [review]
Thanks @JichaoS  and @jmhain

**Pull request merged**